app/actions/Session.js

```diff
@@ -0,0 +1,285 @@
+  return async (dispatch) => {
+    let {name, uuid} = params;
+    dispatch({type: SAVE_SESSION_REQUESTED});
+    let savedSessions = await settings.get(SAVED_SESSIONS) || [];
+    if (!uuid) {
+
+      // If it's a new session, add it to the list
+      uuid = UUID();
+      let newSavedSession = {
+        date: +(new Date()),
+        name,
+        uuid,
+        caps,
+      };
+      savedSessions.push(newSavedSession);
+    } else {
+
+      // If it's an existing session, overwrite it
+      for (let session of savedSessions) {
+        if (session.uuid === uuid) {
+          session.caps = caps;
+        }
+      }
+    }
+    await settings.set(SAVED_SESSIONS, savedSessions);
+    dispatch({type: SET_CAPS, caps, uuid});
+    dispatch({type: SAVE_SESSION_DONE});
+    await getSavedSessions()(dispatch);
+  };
+}
+
+/**
+ * Get the sessions saved by the user
+ */
+export function getSavedSessions () {
+  return async (dispatch) => {
+    dispatch({type: GET_SAVED_SESSIONS_REQUESTED});
+    let savedSessions = await settings.get(SAVED_SESSIONS) || [];
+    dispatch({type: GET_SAVED_SESSIONS_DONE, savedSessions});
+  };
+}
+
+/**
+ * Switch to a different tab
+ */
+export function switchTabs (key) {
+  return (dispatch) => {
+    dispatch({type: SWITCHED_TABS, key});
+  };
+}
+
+/**
+ * Open a 'Save As' modal
+ */
+export function requestSaveAsModal () {
+  return (dispatch) => {
+    dispatch({type: SAVE_AS_MODAL_REQUESTED});
+  };
+}
+
+/**
+ * Hide the 'Save As' modal
+ */
+export function hideSaveAsModal () {
+  return (dispatch) => {
+    dispatch({type: HIDE_SAVE_AS_MODAL_REQUESTED});
+  };
+}
+
+/**
+ * Set the text to save capabilities as
+ */
+export function setSaveAsText (saveAsText) {
+  return (dispatch) => {
+    dispatch({type: SET_SAVE_AS_TEXT, saveAsText});
+  };
+}
+
+/**
+ * Delete a saved session
+ */
+export function deleteSavedSession (index) {
+  return async (dispatch) => {
+    dispatch({type: DELETE_SAVED_SESSION_REQUESTED, index});
+    let savedSessions = await settings.get(SAVED_SESSIONS) || [];
+    savedSessions.splice(index, 1);
+    await settings.set(SAVED_SESSIONS, savedSessions);
+    dispatch({type: GET_SAVED_SESSIONS_DONE, savedSessions});
+  };
+}
+
+/**
+ * Change the server type
+ */
+export function changeServerType (serverType) {
+  return (dispatch) => {
+    dispatch({type: CHANGE_SERVER_TYPE, serverType});
+  };
+}
+
+/**
+ * Set a server parameter (host, port, etc...)
+ */
+export function setServerParam (name, value) {
+  return (dispatch, getState) => {
+    dispatch({type: SET_SERVER_PARAM, serverType: getState().session.serverType, name, value});
+  };
+}
+
+/**
+ * Set the local server hostname and port to whatever was saved in 'actions/StartServer.js' so that it
+ * defaults to what the currently running appium server is
+ */
+export function setLocalServerParams () {
+  return async (dispatch) => {
+    let port = await settings.get('SERVER_PORT');
```

Comment:
If the exception isn't caught, it will generate an unhandled rejection warning, which in later versions of Node 7 will kill the Node process. That would be a bad user experience :-)